src/client/http_handler.rs

```diff
@@ -206,4 +208,102 @@ impl HTTPHandler {
         self.http_do(uri, HttpVerb::Put, payload, headers, deadline, ctrl_c)
             .await
     }
+
+    // TO DO - add ctrlc cancellation
+    pub(crate) async fn http_post_stream(
+        &self,
+        uri: &str,
+        payload: Option<Vec<u8>>,
+        headers: HashMap<&str, &str>,
+        deadline: Instant,
+        ctrl_c: Arc<AtomicBool>,
+    ) -> Result<
+        (
+            impl Stream<Item = Result<Bytes, reqwest::Error>> + Sized,
+            u16,
+        ),
+        ClientError,
+    > {
+        self.http_do_stream(uri, HttpVerb::Post, payload, headers, deadline, ctrl_c)
+            .await
+    }
+
+    pub async fn http_do_stream(
```

Comment:
This is another case where it seems like we don't need both http_do and http_do_stream and http_do just streams anyway. Either post, put, get etc... should also return a stream or we leave post_stream in and the others just consume the stream before returning.